### ISSUE TITLE
Getting the value of RMSE in each iteration

### ISSUE THREAD

**Issue by hdi-amiri** [guoguibing/librec#160]
I want to know the value of `RMSE` in each iteration over the whole dataset, because in most papers I see that the authors show a plot with the value of `RMSE` in each iteration and it converges after 30 to 100 iterations.

in librec the condition for stopping the learning process is evaluating difference between value of loss function after each iteration, in my experiments with librec I saw that PMF on an example dataset didn't converge after 3000 iterations !

How can I stop iterating over the dataset if the change in RMSE value of training data is less than 1e-3 ?
How can I print the value of RMSE for both test set and train set after each iteration?

**Comment by vgeek-z** [comment]
@ha0x7c7 If you understand the function of the method of isConverged(int iter), you can overrider the condition of stopping the learning process. Maybe you can refer:
You can calculate the RMSE in each iteration by changing the loss function to RMSE calculation, then you pass the RMSE to isConverged() method and decide whether stop the learning process.
